tests/integration/models/multifidelity/test_multifidelity_models.py

```diff
@@ -0,0 +1,574 @@
+    initial_sample = tf.concat(initial_samples_list, 0)
+    observer = mk_observer(noisy_linear_multifidelity)
+    initial_data = observer(initial_sample)
+
+    data = split_dataset_by_fidelity(initial_data, n_fidelities)
+
+    gprs = [
+        GaussianProcessRegression(
+            build_gpr(
+                data[fidelity], input_search_space, likelihood_variance=1e-6, kernel_priors=False
+            )
+        )
+        for fidelity in range(n_fidelities)
+    ]
+
+    model = MultifidelityAutoregressive(gprs)
+
+    model.update(initial_data)
+    model.optimize(initial_data)
+
+    test_xs = tf.linspace(0, 1, 11)[:, None]
+    test_xs_w_fid = add_fidelity_column(test_xs, fidelity=3)
+
+    predictions = model.predict(test_xs_w_fid)[0]
+    gt_obs = observer(test_xs_w_fid).observations
+
+    npt.assert_allclose(predictions, gt_obs, rtol=0.20)
+
+
+def test_multifidelity_autoregressive_gets_expected_rhos() -> None:
+
+    input_dim = 1
+    lb = np.zeros(input_dim)
+    ub = np.ones(input_dim)
+    n_fidelities = 4
+
+    input_search_space = trieste.space.Box(lb, ub)
+    n_samples_per_fidelity = [
+        2 ** ((n_fidelities - fidelity) + 1) + 3 for fidelity in range(n_fidelities)
+    ]
+
+    xs = [tf.linspace(0, 1, samples)[:, None] for samples in n_samples_per_fidelity]
+    initial_samples_list = [tf.concat([x, tf.ones_like(x) * i], 1) for i, x in enumerate(xs)]
+    initial_sample = tf.concat(initial_samples_list, 0)
+    observer = mk_observer(noisy_linear_multifidelity)
+    initial_data = observer(initial_sample)
+
+    model = MultifidelityAutoregressive(
+        build_multifidelity_autoregressive_models(initial_data, n_fidelities, input_search_space)
+    )
+
+    model.update(initial_data)
+    model.optimize(initial_data)
+
+    expected_rho = [1.0] + [(fidelity + 1) / fidelity for fidelity in range(1, n_fidelities)]
+    rhos = [float(rho.numpy()) for rho in model.rho]
+
+    npt.assert_allclose(np.array(expected_rho), np.array(rhos), rtol=0.30)
+
+
+def test_multifidelity_autoregressive_predict_low_fidelity_are_consistent() -> None:
+
+    xs_low = tf.Variable(np.linspace(0, 10, 100), dtype=tf.float64)[:, None]
+    xs_high = tf.Variable(np.linspace(0, 10, 10), dtype=tf.float64)[:, None]
+    lf_obs = tf.sin(xs_low)
+    hf_obs = 2 * tf.sin(xs_high) + tf.random.normal(
+        xs_high.shape, mean=0, stddev=1e-1, dtype=tf.float64
+    )
+
+    lf_query_points = add_fidelity_column(xs_low, 0)
+    hf_query_points = add_fidelity_column(xs_high, 1)
+
+    lf_dataset = Dataset(lf_query_points, lf_obs)
+    hf_dataset = Dataset(hf_query_points, hf_obs)
+
+    dataset = lf_dataset + hf_dataset
+
+    search_space = Box([0.0], [10.0])
+
+    model = MultifidelityAutoregressive(
+        build_multifidelity_autoregressive_models(
+            dataset, num_fidelities=2, input_search_space=search_space
+        )
+    )
+
+    model.update(dataset)
+    model.optimize(dataset)
```

Comment:
for this test and all the others: there is a small risk that when calling `optimize` the rho value becomes (or stay) 0 or 1, which weakens the tests. Could we set it to any random value after optimize instead? 

actually we could even get rid of `optimize`?